Include/internal/pycore_bytesobject.h

```diff
@@ -33,6 +34,56 @@ _PyBytes_ReverseFind(const char *haystack, Py_ssize_t len_haystack,
                      const char *needle, Py_ssize_t len_needle,
                      Py_ssize_t offset);
 
+
+/** Helper function to implement the inplace repeat method on a buffer
+ *
+ * len_dest is assumed to be an integer multiple of len_src.
+ *
+ * This method repeately doubles the number of bytes copied to reduce
+ * the number of invocations of memcpy
+ */
+static inline void
+_PyBytes_RepeatInPlace(char* buffer, Py_ssize_t start_len, Py_ssize_t end_len)
```

Comment:
No need for `static inline`. Declare these as `extern void` and put the implementations in Objects/bytesobject.c.

Inlining isn't always the right way to go: inlining makes the most impact when the functions are small and take only a few clock cycles and are called in a tight loop, and so you can shave off the overhead of a few extra clock cycles by removing the function call. These functions, indirectly through mempcy, can take millions of clock cycles, and always take enough clock cycles to make make sub-calls to memcpy.

The proliferation of inlining across a code base can also inflate binary executable sizes, which can hurt instruction cache performance. Besides, compilers with Profile-Guided-Optimization are smart enough to infer which calls ought to be inlined, even if things aren't declared as `inline`.